app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Helper methods for session handling
+module SessionsHelper
+  def log_in(token)
+    cookies.permanent[:remember_token] = token
+  end
+
+  def log_out
+    cookies.delete(:remember_token)
+    self.current_user = nil
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def digest(token)
+    Digest::SHA1.hexdigest(token)
+  end
+
+  def current_user
+    token = cookies[:remember_token]
+    return nil if token.nil?
+
+    @current_user ||= User.find_by(remember_digest: digest(token))
+  end
+
+  def current_user=(user)
+    @current_user = user
+  end
+
+  def require_login
```

Comment:
This should be in the Application controller.

Reply:
That is a method that the view uses to check if the user is logged in. It's fine where it is.

Reply:
I apologize, you can leave this here